run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := []string{}
```

Comment:
Ok, is it also better for the next line to be 'seen := make(map[string]string)' as opposed to 'seen := map[string]string{}'? Maybe this is purely a stylistic difference.